url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgSample.java

```diff
@@ -0,0 +1,998 @@
+                }
+            }
+        } 
+    }
+    
+    private boolean testCommandLine (String commandLine) {
+        clearArgumentVariables();
+        System.out.println("\n" + commandLine);
+        Debug.trace("=========================");
+        String args[] = commandLine.split("\\s");
+        int nArg = argFace.parse(args);
+        if (nArg < 0) {
+            System.out.println("Error: nArg = " + nArg);
+            if (show) {
+                showResults(0, args);
+            }
+            return false;
+        }
+        if (argFace instanceof ArgProcedure) {
+            setArgumentVariables(argFace);
+        }
+        if (show) {
+            showResults(nArg, args);
+        }
+        return true;
+    }
+    
+    private void checkValues (String valueCheck) {
+        if (valueCheck.equals("fail")) {
+            System.out.println("*** ERROR *** expected FAIL didn't happen");
+            return;
+        }
+        String [] tokens = valueCheck.split("\\s");
+        for (String tok : tokens) {
+            checkValue(tok);
+        }
+    }
+    
+    private void checkValue (String tok) {
+        String [] nvs = tok.split("=", 2);
+        String name = nvs[0];
+        String value = null;
+        if (nvs.length == 2) {
+            value = nvs[1];
+        }
+        if (! argFace.has(name)) {
+            System.out.printf("*** ERROR ***  %s fails \"has\", should be true%n", name);
+        }
+        if (! checkTrue(name)) {
+            System.out.printf("*** ERROR ***  %s program variable should be set%n", name);
+        }
+        if (value != null) {
+            if (value.startsWith(":")) {
+                value = value.substring(1);
+                int valCount = intPositive(value);
+                int count = argFace.count(name);
+                if (valCount != count) {
+                    System.out.printf("*** ERROR ***  %s count expected %d is %d%n",
+                            name, valCount, count);
+                }
+                return;
+            } else if (value.startsWith("[")) {
+                value = value.substring(1);
+                String[] values = value.split(",");
+                String[] vals = argFace.valueArray(name);
+                for (int n = 0; n < vals.length; n++) {
+                    value = values[n];
+                    if (! value.equals(vals[n])) {
+                        System.out.printf("*** ERROR ***  " +
+                                "%s value expected \"%s\" is \"%s\"%n,",
+                                name, value, vals[n]);
+                    }
+                }
+            }
+            String val = argFace.value(name);
+            if (! value.equals(val)) {
+                System.out.printf("*** ERROR ***  %s value expected \"%s\" is \"%s\"%n",
+                        name, value, val);
+            }
+        }
+    }
+    
+    private boolean checkTrue (String name) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 30. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.